modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/organization/management/v1/OrganizationManagementSuccessTest.java

```diff
@@ -604,4 +608,200 @@ private void deleteApplication(String applicationId) throws Exception {
+        after = getLink(secondPageLinks, "next");
+
+        Assert.assertNotNull(before, "Before value should not be null on the second page.");
+        if (totalCreatedOrganizations > limit * 2) {
+            Assert.assertNotNull(after, "After value should not be null if there are more pages.");
+        } else {
+            Assert.assertNull(after, "After value should be null if this is the last page.");
+        }
+
+        // Step 3: Call the previous page using the 'before' value
+        String previousPageUrl = ORGANIZATION_MANAGEMENT_API_BASE_PATH + LIMIT_ARG + limit
+                + "&recursive=false&before=" + before;
+        Response previousPageResponse = getResponseOfGetWithOAuth2(previousPageUrl, m2mToken);
+
+        validateHttpStatusCode(previousPageResponse, HttpStatus.SC_OK);
+
+        int previousPageOrgCount = previousPageResponse.jsonPath().getList("organizations").size();
+        Assert.assertEquals(previousPageOrgCount, firstPageOrgCount);
+
+        // Ensure that the organizations match the first page organizations
+        Assert.assertEquals(previousPageResponse.jsonPath().getList("organizations"),
+                firstPageResponse.jsonPath().getList("organizations"),
+                "The previous page should return the same organizations as the first page.");
+
+        // Extract only the name and description from the previousPageResponse for comparison
+        List<Map<String, String>> actualOrganizations = previousPageResponse.jsonPath().getList("organizations");
+
+        List<Map<String, String>> actualOrganizationsSimplified = new ArrayList<>();
+        for (Map<String, String> org : actualOrganizations) {
+            Map<String, String> simplifiedOrg = new HashMap<>();
+            simplifiedOrg.put(ORGANIZATION_NAME, org.get(ORGANIZATION_NAME));
+            simplifiedOrg.put(ORGANIZATION_DESCRIPTION, "Test " + org.get(ORGANIZATION_NAME));
+            actualOrganizationsSimplified.add(simplifiedOrg);
+        }
+
+        // Additional validation: Check that the previousPageResponse contains the last 'limit' organizations
+        List<Map<String, String>> expectedLastOrganizations =
+                createdOrganizations.subList(totalCreatedOrganizations - limit, totalCreatedOrganizations);
+        Collections.reverse(expectedLastOrganizations); // Reverse the order to match the expected response
+        Assert.assertEquals(actualOrganizationsSimplified, expectedLastOrganizations,
+                "The previous page should return the last created organizations.");
+
+
+        // Step 4: Test with orgLimit (equal to total org count)
+        validatePaginationForNonPaginatedLimit(orgLimit);
+
+        // Step 5: Test with largeLimit (greater than total org count)
+        validatePaginationForNonPaginatedLimit(largeLimit);
+    }
+
+    private int createOrganizations(int numberOfOrganizations) throws JSONException {
+
+        createdOrganizations = new ArrayList<>();
```

Comment:
Let's create a local organizations list and return it from this private method.

Reply:
Done